.github/workflows/ci-concurrency.yml

```diff
@@ -0,0 +1,40 @@
+name: Concurrency Tests
+
+on:
+  push:
+    branches: [ 'main*' ]
+    paths-ignore:
+    - '**.md'
```

Comment:
Since the eventual goal is to enforce concurrency tests in CI, need to add `ci-concurrency-md.yml` similar to other `*-md.yml` files under https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/.github/workflows.

Reply:
@Yun-Ting Need to add one for AOT as well.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4938

Reply:
@CodeBlanch is bringing in CI improvements to this repo similar to: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1383.
(Let's see which PR got merged first 😏.)

Reply:
Yes I plan to do away with the need for these "-md" workflows soon. Next day or two, just need to find some free cycles 😄

Reply:
I've added a -md file for the coyote flow just now!